guidelines/goals.md

```diff
@@ -0,0 +1,71 @@
+# Goals and Non-Goals
+
+This document defines the purpose of the Message Format Working Group (MFWG)
+and informs the decisions about the scope and the priorities of its efforts.
+
+## Goals
+
+The primary task of the MFWG is to develop an industry standard for the
+representation of localizable message strings. The design goals are listed
+below.
+
+ 1. Express complex grammatical features, such as plurals, genders, and
+    inflections, catalogued in the CLDR.
+
+ 2. Express other variance in translation, due to other linguistic and regional
+    features not listed in the CLDR, as well as due to the presentation media,
+    context, circumstance, and other factors.
+
+ 3. Represent structured data alongside translations, like markup, comments,
+    and metadata.
+
+ 4. Ensure interoperability with existing interchange formats, in particular
+    with XLIFF.
+
+ 5. Ensure that machine translation and translation memory solutions can be
+    used for authoring translations.
+
+ 6. Enable a wide range of user space solutions building on top of the
+    standard, manifesting different ideas and programming paradigms, and
+    optimized for different uses and audiences.
+
+
+## Deliverables
+
+ 1. A formal definition of the canonical data model for representing
+    translations.
+
+ 2. A formal definition of the canonical syntax for representing the data
+    model, with well defined rules for handling text, special characters,
+    escape sequences, whitespace, markup, as well as parsing errors.
+
+ 3. A specification for lossless conversion between the data model and XLIFF.
+
```

Comment:
IIUC, in the l10n tools world, XLIFF is the de facto standard format that they all support for importing/exporting, regardless of which other file formats they support.  It's an open standard that is not vendor-specific.  The core of it is implemented across most tools, even if some may attempt an embrace,extend,... strategy.

Reply:
I'd argue that XLIFF is probably the most complete (and complex) standard that is also not consistently adopted to the same extent across many different products. It might be interesting to also consider the opposite file format (the simplest), which consist of only `keys` and `values` - for example `.properties` file. If we could support both then we should have a complete and flexible solution.

Reply:
This is a great point. It has been my experience as well that many vendors say they support XLIFF, and then it turns out that everyone supports a different subset of XLIFF.